test/distributed_test.rb

```diff
@@ -1,56 +1,65 @@
-require_relative "helper"
+# frozen_string_literal: true
 
-class TestDistributed < Test::Unit::TestCase
+require_relative 'helper'
 
+class TestDistributed < Test::Unit::TestCase
   include Helper::Distributed
 
   def test_handle_multiple_servers
-    @r = Redis::Distributed.new ["redis://127.0.0.1:#{PORT}/15", *NODES]
-
     100.times do |idx|
-      @r.set(idx.to_s, "foo#{idx}")
+      r.set(idx.to_s, "foo#{idx}")
     end
 
     100.times do |idx|
-      assert_equal "foo#{idx}", @r.get(idx.to_s)
+      assert_equal "foo#{idx}", r.get(idx.to_s)
     end
 
-    assert_equal "0", @r.keys("*").sort.first
-    assert_equal "string", @r.type("1")
+    assert_equal '0', r.keys('*').min
+    assert_equal 'string', r.type('1')
   end
 
   def test_add_nodes
-    logger = Logger.new("/dev/null")
+    logger = Logger.new('/dev/null')
 
-    @r = Redis::Distributed.new NODES, :logger => logger, :timeout => 10
+    r = build_another_client(distributed: { nodes: NODES }, logger: logger, timeout: 10)
 
-    assert_equal "127.0.0.1", @r.nodes[0]._client.host
-    assert_equal PORT, @r.nodes[0]._client.port
-    assert_equal 15, @r.nodes[0]._client.db
-    assert_equal 10, @r.nodes[0]._client.timeout
-    assert_equal logger, @r.nodes[0]._client.logger
+    assert_equal '127.0.0.1', r.nodes[0].host
+    assert_equal PORT, r.nodes[0].port
+    assert_equal DB, r.nodes[0].db
+    assert_equal 10, r.nodes[0].timeout
+    assert_equal logger, r.nodes[0].logger
 
-    @r.add_node("redis://127.0.0.1:6380/14")
+    r.add_node("redis://127.0.0.1:#{PORT}/14")
 
-    assert_equal "127.0.0.1", @r.nodes[1]._client.host
-    assert_equal 6380, @r.nodes[1]._client.port
-    assert_equal 14, @r.nodes[1]._client.db
-    assert_equal 10, @r.nodes[1]._client.timeout
-    assert_equal logger, @r.nodes[1]._client.logger
+    assert_equal '127.0.0.1', r.nodes[1].host
+    assert_equal NODE2_PORT, r.nodes[1].port
+    assert_equal DB, r.nodes[1].db
+    assert_equal 10, r.nodes[1].timeout
+    assert_equal logger, r.nodes[1].logger
   end
 
   def test_pipelining_commands_cannot_be_distributed
     assert_raise Redis::Distributed::CannotDistribute do
       r.pipelined do
-        r.lpush "foo", "s1"
-        r.lpush "foo", "s2"
+        r.lpush 'foo', 's1'
+        r.lpush 'foo', 's2'
       end
     end
   end
 
   def test_unknown_commands_does_not_work_by_default
-    assert_raise NoMethodError do
+    assert_raise Redis::CommandError do
       r.not_yet_implemented_command
     end
   end
+
+  def test_backward_compatible_class
+    rd = Redis::Distributed.new(NODES, timeout: TIMEOUT)
+    rd.set('foo', '1')
+    assert_equal '1', rd.get('foo')
+
+    hr = Redis::HashRing.new
```

Comment:
This is used by some gems. e.g. [redis-store](https://github.com/redis-store/redis-store/blob/v1.6.0/lib/redis/distributed_store.rb#L10)